jb/itch5/order_book_depth.hpp

```diff
@@ -0,0 +1,175 @@
+#ifndef jb_itch5_order_book_depth_hpp
+#define jb_itch5_order_book_depth_hpp
+
+#include <jb/itch5/price_field.hpp>
+#include <jb/itch5/buy_sell_indicator.hpp>
+#include <jb/log.hpp>
+
+#include <functional>
+#include <map>
+#include <utility>
+
+namespace jb {
+namespace itch5 {
+
+  // To keep the depth of book per symbol
+typedef unsigned long int book_depth_t;   
+
+/**
+ * Maintain the ITCH-5.0 order book for a single security.
+ *
+ * ITCH-5.0, like other market data feeds provide order-by-order
+ * detail, that is, the feed includes a message for each order
+ * received by the exchange, as well as the changes to these 
+ * orders, i.e. when the execute, when their quantity (and/or price)
+ * is modified, and when they are canceled..  Such feeds are sometimes
+ * called Level III feeds.  Typically only orders that do not
+ * immediately execute in full are included in the feed.
+ *
+ * There is substantial variation in the format of the messages, and
+ * some variation as to whether executions are represented differently
+ * than a partial cancel, and whether changes in price are allowed or
+ * create new order ids.
+ *
+ * This class receives a stream of (unnormalized) ITCH-5.0 messages
+ * for a single security, and organizes the orders in a
+ * book, i.e. a  data structure where orders at the same price are
+ * grouped together, and one can quickly ask:
+ *
+ * - What is the best bid (highest price of BUY orders) and what is
+ * the total quantity available at that price?
+ * - What is the best offer (lowest price of SELL orders) and what is
+ * the total quantity available at that price?
+ */
+class order_book_depth {
+ public:
+  /// Initialize an empty order book.
+  order_book_depth() {}
+
+  /* Ticket https://github.com/coryan/jaybeams/issues/20
+   * 
+   * Return the book_depth on this order_book  
+   * @RETURN : const reference to book_depth_
+   */
+  const book_depth_t& get_book_depth() const {return book_depth_;};
```

Comment:
For integral types returning by value is more idiomatic (and arguably more efficient), please consider fixing the code to simply return the value.